### PR TITLE
Convert to KMP

### PR DESCRIPTION
**Pull request by alesrazym**

### CODE REVIEW

File: android/app/src/main/java/cz/quanti/rocketapp/di/RocketModule.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package cz.quanti.rocketapp.di
+
+import androidx.navigation.navOptions
+import cz.quanti.rocketapp.R
+import cz.quanti.rocketapp.presentation.RocketDetailViewModel
+import cz.quanti.rocketapp.presentation.RocketListViewModel
+import org.koin.androidx.viewmodel.dsl.viewModelOf
+import org.koin.dsl.module
+
+val rocketModule = module {
```
Comment (havlisimo) [inline android/app/src/main/java/cz/quanti/rocketapp/di/RocketModule.kt]: rocket module by mel byt v rocket gradle modulu

File: android/app/src/main/java/cz/quanti/rocketapp/system/RocketLaunchScreen.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.compose.composable
+import cz.quanti.rocketapp.R
+import cz.quanti.rocketapp.ui.TopBarWithBackIcon
+import cz.quanti.rocketapp.ui.theme.RocketappTheme
+
+data object RocketLaunchScreen : Screen("rocketLaunch")
```
Comment (havlisimo) [inline android/app/src/main/java/cz/quanti/rocketapp/system/RocketLaunchScreen.kt]: screeny by mely byt take v rocket modulu, tady v app by melo zustat jen to co to vsechno spojuje dohromady (App implementace, main activity, start koin, navigace atd.)

File: android/app/src/main/java/cz/quanti/rocketapp/di/RocketModule.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    single { provideNavOptions() }
+    viewModelOf(::RocketListViewModel)
+    viewModelOf(::RocketDetailViewModel)
+}
+
+private fun provideNavOptions() = navOptions {
```
Comment (havlisimo) [inline android/app/src/main/java/cz/quanti/rocketapp/di/RocketModule.kt]: toto nejspis v UI modulu

File: android/app/src/main/res/drawable/burn.xml
Comment (havlisimo) [inline android/app/src/main/res/drawable/burn.xml:1]: resources tykajici se rocket obrazovek take do rocket modulu

File: android/rocket/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    namespace = "cz.quanti.rocketapp.android.rocket"
+
+    defaultConfig {
+        consumerProguardFiles("consumer-rules.pro")
+    }
+
+    buildTypes {
+        release {
+        }
+    }
```
Comment (havlisimo) [inline android/rocket/build.gradle.kts]: je tento blok potreba?

File: android/rocket/proguard-rules.pro [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (havlisimo) [inline android/rocket/proguard-rules.pro]: proguard-rules.pro nechat jen v app a v ostatnich modulech nechat ten consumer-rules.pro

File: android/rocket/build.gradle.kts (hunk 4 above)
Comment (alesrazym) [reply android/rocket/build.gradle.kts]: neni, je definovano v convention pluginu

File: android/rocket/proguard-rules.pro (hunk 5 above)
Comment (alesrazym) [reply android/rocket/proguard-rules.pro]: koukal jsem se na nastaveni, snad je to ted v poradku a v convention pluginu.

**Comment by alesrazym** [comment]
@havlisimo Budeme chtit nastavit proguard? Ted to neni nastaveno a nefunguje build, pokud se zapne.

**Comment by alesrazym** [comment]
I've rebased branch and merged fixes and rfc from #21. There's `AppModule` added to be the only module to be used in app and in tests.

**Review by havlisimo — APPROVED** [review]
iOS se mi otestovat nepodarilo, nicmene kod vypada celkem v cajku, nekde jeste nesedi clean architektura, ale to bych resil v druhem PR ohledne konsist.

**Pull request merged**